pkg/skaffold/docker/parse.go

```
@@ -198,21 +198,66 @@ func readDockerfile(workspace, absDockerfilePath string, buildArgs map[string]*s
 		return nil, errors.Wrap(err, "parsing dockerfile")
 	}
 
-	expandBuildArgs(res.AST.Children, buildArgs)
+	dockerfileLines := res.AST.Children
 
-	instructions, err := onbuildInstructions(res.AST.Children)
+	expandBuildArgs(dockerfileLines, buildArgs)
+
+	dockerfileLines, err = forTarget(target, dockerfileLines)
+	if err != nil {
+		return nil, errors.Wrap(err, "filtering on target")
+	}
+
+	instructions, err := onbuildInstructions(dockerfileLines)
 	if err != nil {
 		return nil, errors.Wrap(err, "listing ONBUILD instructions")
 	}
 
-	copied, err := copiedFiles(append(instructions, res.AST.Children...))
+	copied, err := copiedFiles(append(instructions, dockerfileLines...))
 	if err != nil {
 		return nil, errors.Wrap(err, "listing copied files")
 	}
 
 	return expandPaths(workspace, copied)
 }
 
+func forTarget(target string, nodes []*parser.Node) ([]*parser.Node, error) {
+	if target == "" {
+		return nodes, nil
+	}
+
+	byTarget := make(map[string][]*parser.Node)
```

`var byTarget map[string][]*parser.Node`

---

I don't think so. Setting a value on a nil map fails hard